packages/common/core-interfaces/src/brandedType.ts

```diff
@@ -7,14 +7,55 @@
  * Base branded type which can be used to annotate other type.
  *
  * @remarks
- * To use derive another class declaration and ideally add additional private
- * properties to further distinguish the type.
+ * To use, derive another class declaration and ideally add additional private
+ * and/or protected (generally preferred) properties to distinguish the type.
+ * (Any private properties should be typed as `unknown` as external to module
+ * the only thing known about a private property is that it exists.)
+ * A private constructor is recommended to seal the brand class.
```

Comment:
I'd be more strict about how this is used.

I don't think we need to consider non-ideal usage as valid when there is no reason to not just improve it.
Taking such a strict approach avoids people making tradoff judgments on if the non ideal usage makes sense when there is no actual reason to use the non ideal approach, and the issues with it are subtle.

```suggestion
 * To use, derive another class declaration and add additional protected properties to distinguish the type.
 * Private properties would {@link https://github.com/microsoft/TypeScript/issues/20979#issuecomment-361432516|lose their type when exported} and public ones would allow structural typing and show up on the branded values.
 *
 * A private constructor is recommended to seal the brand class.
```